docs/BUILDER_INTEGRATION.md

```diff
@@ -1,7 +1,504 @@
+
+The flow-builder can be installed as a package from npm:
+
+```
+yarn add @floip/flow-builder
+
+or
+
+npm install --save @floip/flow-builder
+```
+
+Unlike the standalone app, the built version of the flow-builder in `/dist` is intended to integrate with an existing Vue Router setup instead of using the exact routing config in https://github.com/FLOIP/flow-builder/blob/master/src/router/trees.js. 
+
+First we need some config to let the components know what routes are available. For example we could have:
+
+```
+const routes = {
+    "flows": {
+        "persistFlow": {
+            "name": "persistFlow",
+            "path": "/backend/flows",
+            "methods": [
+                "POST",
+                "PUT"
+            ]
+        },
+        "fetchFlowServer": {
+            "name": "fetchFlowServer",
+            "path": "/backend/flows/{flowId}",
+            "params": [
+                "flowId"
+            ],
+            "methods": [
+                "GET"
+            ]
+        },
+        "newFlow": {
+            "name": "newFlow",
+            "path": "/flows/new",
+            "methods": [
+                "GET"
+            ]
+        },
+        "importFlow": {
+            "name": "importFlow",
+            "path": "/flows/import",
+            "methods": [
+                "GET"
+            ]
+        },
+        "fetchFlow": {
+            "name": "fetchFlow",
+            "path": "/flows/{flowId}",
+            "params": [
+                "flowId"
+            ],
+            "methods": [
+                "GET"
+            ]
+        },
+        "home": {
+            "name": "home",
+            "path": "/host-app/flows",
+            "methods": [
+                "GET"
+            ]
+        },
+        "cancelImport": {
+            "name": "home",
+            "path": "/flows",
+            "methods": [
+                "GET"
+            ]
+        },
+        "editFlow": {
+            "name": "editFlow",
+            "path": "/flows/{flowId}/{mode?}",
+            "params": [
+                "flowId",
+                "component",
+                "mode"
+            ],
+            "methods": [
+                "GET",
+                "HEAD"
+            ]
+        }
+    },
+    "trees": {
+        "resumeableAudioUpload": {
+            "name": "resumeableAudioUpload",
+            "path": "/audiofiles/upload",
+            "params": [],
+            "methods": [
+                "POST",
+                "HEAD"
+            ]
+        }
+    }
+}
+
+```
+
+And override the builder.config.json defaults from the community builder like so:
+
+```
+//Importing default config
+import {
+  builderConfig as platformBuilderConfig,
+} from '@floip/flow-builder'
+
+platformBuilderConfig.ui.routes = routes
+
+```
+Then we can configure Vue Router for this. Note passing in of `platformBuilderConfig` to make components aware of the configured routes:
+
+```
+
+const flowBuilderRoutes [
```

Comment:
Hmm, is this definitely something we want in this case considering we're trying to preserve flexibility and allow fitting the builder into the routing config of another app? We have approached things this way to avoid having to export routing config in flow-builder and consume it in voto5. 

Or are you suggesting we also introduce nuxt in voto5 and use this auto generation there (but a different directory structure to achieve what we want?)

Reply:
hum, I haven't thought about that yet, but you might be right. Never mind, let's ignore that for now, we have lots of spike to do :D